src/classlibnative/bcltype/number.cpp

```diff
@@ -1983,28 +1986,21 @@ STRINGREF NumberToStringFormat(NUMBER* number, STRINGREF str, NUMFMTREF numfmt)
 #pragma warning(pop)
 #endif
 
-FCIMPL3_VII(void, COMNumber::DoubleToNumberFC, double value, int precision, NUMBER* number)
-{
-    FCALL_CONTRACT;
 
-    DoubleToNumber(value, precision, number);
-}
-FCIMPLEND
-
-FCIMPL1(double, COMNumber::NumberToDoubleFC, NUMBER* number)
+#if defined(_MSC_VER)
+FCIMPL6(void, COMNumber::DoubleToStringWindows, char* buffer, int sizeInBytes, double value, int count, int* dec, int* sign)
 {
     FCALL_CONTRACT;
-
-    double d = 0;
-    NumberToDouble(number, &d);
-    return d;
+    _ecvt_s(buffer, sizeInBytes, value, count, dec, sign);
 }
 FCIMPLEND
+#else
 
-FCIMPL2(FC_BOOL_RET, COMNumber::NumberBufferToDecimal, NUMBER* number, DECIMAL* value)
+FCIMPL4(int, COMNumber::DoubleToStringUnix, double value, char *format, char *buffer, int bufferLength)
 {
     FCALL_CONTRACT;
-
-    FC_RETURN_BOOL(COMDecimal::NumberToDecimal(number, value) != 0);
+    int result = sprintf_s(buffer, bufferLength, format, value);
```

Comment:
CoreCLR has optimized implementation of DoubleToNumber (look for Grisu3 and Dragon4). It is a lot faster than the `sprintf_s` or `_ecvt_s`. This change is disabling the optimization.

As was pointed out earlier, the formatting/parsing related changes should be looked as separate change.

Reply:
@jkotas ok, will separate them, however according to my benchmarks (like this:
```
[Benchmark] public decimal Parse5() => decimal.Parse("1024.1024");
etc
```
there is no difference (and even a little bit faster then CoreCLR-master) but I am benchmarking only Windows (probably `sprintf_s` is slower)